segmentation.py

```diff
@@ -0,0 +1,575 @@
+"""Segment time sequence data.
+
+The Bayesian change-point detection model (BCDM) class implements a recursive
+algorithm for partitioning a sequence of real-valued input-output data into
+non-overlapping segments. The segment boundaries are chosen under the
+assumption that, within each segment, the data follow a multi-variate linear
+model.
+
+Segmentation is carried out in an online fashion by recursively updating a set
+of hypotheses. The hypotheses capture the belief about the current segment,
+e.g. its duration and the linear relationship between inputs and outputs, given
+all the data so far. Each time a new pair of data is received, the hypotheses
+are propagated and re-weighted to reflect this new knowledge.
+
+.. codeauthor:: Gabriel Agamennoni <abriel.agamennoni@mavt.ethz.ch>
+.. codeauthor:: Asher Bender <a.bender@acfr.usyd.edu.au>
+
+"""
+import numpy as np
```

Comment:
I'm just paranoid about name clashes. But I can't think of a strong argument not to use the `np` binding.